src/views/Entries/QualityControl/index.tsx

```diff
@@ -319,10 +319,17 @@ function QualityControl(props: Props) {
         handleEntryDelete,
     ]);
 
-    const tocFilterRendererParams = useCallback((_: string, data: MatrixKeyId) => ({
-        className: styles.badge,
-        title: labelSelector(data),
-    }), []);
+    const handleRemoveSelection = useCallback((id: string) => {
+        const newTocFilters = tocFilters.filter(f => idSelector(f) !== id);
+        setTocFilters({ tocFilters: newTocFilters });
```

Comment:
```suggestion
        setTocFilters((oldToCFilters) => oldToCFilters.filter(f => idSelector(f) !== id));
```

Reply:
here setTocFilters is a redux function, not a useState update function.

Reply:
Sounds good!